packages/server/api/webscrape.py

```diff
@@ -3,32 +3,63 @@
 import re
 from urllib import parse
 from bs4 import BeautifulSoup
+import json
 
 class handler(BaseHTTPRequestHandler):
   whitelisted_url_regexes = [
     re.compile('^https:\/\/www\.bonappetit\.com')
   ]
 
+  def error_handler(error_string, error_code = 500):
+    self.send_response(error_code)
+    self.send_header('Content-type', 'text/plain')
+    self.end_headers()
+    self.wfile.write(errror_string.encode())
+
   def do_GET(self):
     url = parse.parse_qs(parse.urlsplit(self.path).query)['url'][0]
     url_in_whitelist = any(regex.match(url) for regex in self.whitelisted_url_regexes)
     
     if(not url_in_whitelist):
-      self.send_response(500)
-      self.send_header('Content-type', 'text/plain')
-      self.end_headers()
-      self.wfile.write('Unsupported recipe site'.encode())
+      error_handler('URL not supported')
+      return
+    
+    recipe_object = {
+      'servings': '',
+      'prepTime': '',
+      'cookTime': '',
+      'ingredients': '',
```

Comment:
Are we planning on properly parsing ingredients later on? Right now we just store them as a string

Reply:
Yeah so we were thinking we would make this into a spike as there is a lot of variability in the way that the ingredient info is displayed on these sites, even between recipes, so it would probably be much easier to just use strings for each line item of the ingredient list

Reply:
I'm okay with leaving it that way for now, but if we want to enable anything related to the discoverability of ingredients or searching for recipes based on what's in your pantry, then this will be a problem

Reply:
https://www.notion.so/How-to-handle-ingredients-lists-that-don-t-have-separated-out-quantities-f1a05b7889bc497aa0281ad4836a7982